Platformio config

hoermto opened the pull request:
wemos_d1_mini and d1_mini_32 build also on windows, no need for arduino install.
    
- using default libraries, where possible
- does not require installation of arduino IDE, just VSCode + platformio plugin
- unused libs will be stripped automatically from binary from compiler, no need to specifically exclude them
- remark: on windows rename the .pio/libdeps/wemos_d1_mini32/Time/time.h (or any ESP32) to _time.h for building
  its seems a problem with the webserver library on windows

Also adds environments for AZ Delivery V4 (ESP32) and NodeMCUV2 (ESP8266) in a separate commit

Builds Tested on Windows and Ubuntu.

hoermto committed:
wemos_d1_mini and d1_mini_32 build also on windows, no need for arduino install.

- using default libraries
- does not require installation of Arduino IDE, just VSCode + platformio plugin
- unused libs will be stripped from binary from compiler, no need to specifically exclude them
remark: on windows rename the .pio/libdeps/wemos_d1_mini32/Time/time.h to _time.h for
  its seems a problem with the AsyncWebserver library on windows-amd64 platforms.

Macrisu commented:
> wemos_d1_mini and d1_mini_32 build also on windows, no need for arduino install.
> 
> * using default libraries, where possible
> * does not require installation of arduino IDE, just VSCode + platformio plugin
> * unused libs will be stripped automatically from binary from compiler, no need to specifically exclude them
> * remark: on windows rename the .pio/libdeps/wemos_d1_mini32/Time/time.h (or any ESP32) to _time.h for building
>   its seems a problem with the webserver library on windows
> 
> Also adds environments for AZ Delivery V4 (ESP32) and NodeMCUV2 (ESP8266) in a separate commit
> 
> Builds Tested on Windows and Ubuntu.

Can you please recompile the source for ESP32?
Thnak you

hoermto commented:
Just did for all 4 platforms successfully. Made some minor changes to bring back the version references for the "official" Wemos boards, and using latest versions for the additional 8266 / ESP32 boards. There is one thing to consider building on Windows (`Time.h`). See `platformio.ini` and Readme
Just checking the recommended ModbusMaster library from this repo to be used in platformio.

hoermto committed:
build support for az-delivery-devkit-v4 (ESP32) and nodemcuv2 (8266).

Uses latest available libraries where possible. Binaries are compatible the Wemos D1
boards. However, use these entries for custom builds to not break the
"official" Wemos based product builds

hoermto committed:
Readme for vscode

hoermto commented:
Ok, just rebuilt all platforms on Windows and Ubuntu successfully, wiping all platformio and build artifacts before.
Also changed to use the `CurtRod/ModbusMaster` library as recommended in vscode.

hoermto committed:
add Pin configuration readme

hoermto committed:
use global variable for libraries
